scripts/mkstatic.py

```diff
@@ -128,11 +129,22 @@ def create_arguments_module_use_var_defs(variable_dictionary, metadata_define, t
                 type_var = metadata_define[type_var_standard_name][0]
                 module_use.append(type_var.print_module_use())
                 local_kind_and_type_vars.append(type_var_standard_name)
+
     # Add any local variables (required for unit conversions, array transformations, ...)
     if tmpvars:
         var_defs.append('! Local variables for unit conversions, array transformations, ...')
         for tmpvar in tmpvars:
             var_defs.append(tmpvar.print_def_local())
+            # Add special kind variables
+            if tmpvar.type in STANDARD_VARIABLE_TYPES and tmpvar.kind and not tmpvar.type == STANDARD_CHARACTER_TYPE:
+                kind_var_standard_name = tmpvar.kind
+                if not kind_var_standard_name in local_kind_and_type_vars:
+                    if not kind_var_standard_name in metadata_define.keys():
+                        raise Exception("Kind {kind} not defined by host model".format(kind=kind_var_standard_name))
+                    kind_var = metadata_define[kind_var_standard_name][0]
+                    module_use.append(kind_var.print_module_use())
+                    local_kind_and_type_vars.append(kind_var_standard_name)
+
```

Comment:
Is this directly related to the new phases? I do not immediately see the connection.

Reply:
Indirectly. By coincidence, all existing phases (`_run`, `_init`, `_finalize`) always had at least one scheme for each group that used variables with a `kind_phys` or `kind_dyn` type directly (not just as part of a DDT_. Because of the lines immediately preceding the new block in lined 138-147, `kind_phys` was getting passed correctly to the auto-generated caps.

For the `time_vary` group and phase `timestep_init`, only the DDTs needed to be passed from the host model to the auto-generated cap, and `kind_phys` was thus not known to the cap. When blocked data conversions happen, the temporary variables for the DDT constituents are of `kind_phys`. Thus the additional logic.